### ISSUE TITLE
Indic font rendering incorrectly

### ISSUE THREAD

**Issue by neevany** [bevyengine/bevy#13010]
## Bevy version

11afe160791cdc8c043afb7a7c68cfd8ae1c6a87 / master

## Relevant system information
```
SystemInfo { os: "MacOS 14.4.1 ", kernel: "23.4.0", cpu: "Apple M2 Pro"}
```
```
AdapterInfo { name: "Apple M2 Pro", vendor: 0, device: 0, device_type: IntegratedGpu, driver: "", driver_info: "", backend: Metal }
```

## What you did
https://fonts.google.com/noto/specimen/Noto+Sans+Telugu
```
let text_style = TextStyle {
    font: asset_server.load("fonts/NotoSansTelugu-Medium.ttf"),
    font_size: 60.0,
    ..default()
};
parent.spawn(TextBundle {
  text: Text {
      sections: vec![TextSection::new("రాష్ట్రముల మధ్య", text_style.clone())],
      ..default()
  },
  ..default()
});
```

## What went wrong
the font is not rendering correctly
- what were you expecting?
   to render font correctly like in the screenshot from google fonts site
<img width="536" alt="Screenshot 2024-04-18 at 8 01 54 AM" src="https://github.com/bevyengine/bevy/assets/18303579/2e061827-3530-4590-83f6-769a14f457f6">

- what actually happened?
it did not render font correctly like in this screenshot
<img width="409" alt="Screenshot 2024-04-18 at 8 00 53 AM" src="https://github.com/bevyengine/bevy/assets/18303579/9eae041f-124a-4ca7-aa10-5a9f0a552fec">

**Comment by jordanhalase** [comment]
I do not know how to fix this but I decomposed the string into its code points and graphemes (according to [unicode-segmentation](https://crates.io/crates/unicode-segmentation)).

```
రాష్ట్రముల మధ్య

Code points:

U+0C30		ర
U+0C3E		ా
U+0C37		ష
U+0C4D		్
U+0C1F		ట
U+0C4D		్
U+0C30		ర
U+0C2E		మ
U+0C41		ు
U+0C32		ల
U+0020		 
U+0C2E		మ
U+0C27		ధ
U+0C4D		్
U+0C2F		య

Graphemes (unicode-segmentation): 10

రా
ష్
ట్
ర
ము
ల
 
మ
ధ్
య
```

The first character looks like a ర with a ◌ా nonspacing mark (U+0C3E) appended. The combining mark is tacked on, but the glyph should render differently when it is applied.

The next visual glyph appears to be made up of five code points, which render completely differently if any are removed. This is me pressing backspace on the second glyph:

ష్ట్ర ష్ట్ ష్ట ష్ ష

This definitely isn't an easy problem to solve especially with the way Bevy texture atlases fonts. Does anyone have any thoughts?

**Comment by rparrett** [comment]
Looks like this was fixed by #10193

<img width="429" alt="image" src="https://github.com/user-attachments/assets/e716997a-e4bb-4d52-8365-6fa5d8de7aea">
